R/derive_param_merged_exist_flag.R

```diff
@@ -0,0 +1,237 @@
+#'   the subject at least one observation exists in the (restricted) additional
+#'   dataset where the condition evaluates to `TRUE`. It is set to the false
+#'   value (`false_value`) if for the subject at least one observation exists
+#'   and for all observations the condition evaluates to `FALSE` or `NA`.
+#'   Otherwise, it is set to the missing value (`missing_value`), i.e., for
+#'   those subject not in `dataset_add`.
+#'
+#'   1. The `AVAL` variable is added and set to `aval_fun(AVALC)`.
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @return The input dataset with a new parameter indicating if and when an
+#'   event occurred
+#'
+#' @keywords derivation bds
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#'
+#' # Derive a new parameter for the first disease progression (PD)
+#' adsl <- tibble::tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3"
+#' ) %>%
+#'   mutate(STUDYID = "XX1234")
+#'
+#' tu <- tibble::tribble(
+#'   ~USUBJID, ~VISIT,      ~TUSTRESC,
+#'   "1",      "SCREENING", "TARGET",
+#'   "1",      "WEEK 1",    "TARGET",
+#'   "1",      "WEEK 5",    "TARGET",
+#'   "1",      "WEEK 9",    "NON-TARGET",
+#'   "2",      "SCREENING", "NON-TARGET",
+#'   "2",      "SCREENING", "NON-TARGET"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "XX1234",
+#'     TUTESTCD = "TUMIDENT"
+#'   )
+#'
+#' derive_param_merged_exist_flag(
+#'   dataset_adsl = adsl,
+#'   dataset_add = tu,
+#'   filter_add = TUTESTCD == "TUMIDENT" & VISIT == "SCREENING",
+#'   condition = TUSTRESC == "TARGET",
+#'   false_value = "N",
+#'   missing_value = "N",
+#'   set_values_to = vars(
+#'     PARAMCD = "MDIS",
+#'     PARAM = "Measurable Disease at Baseline"
+#'   )
+#' )
+derive_param_merged_exist_flag <- function(dataset = NULL,
+                                           dataset_adsl,
+                                           dataset_add,
+                                           condition,
+                                           true_value = "Y",
+                                           false_value = NA_character_,
+                                           missing_value = NA_character_,
+                                           filter_add = NULL,
+                                           aval_fun = yn_to_numeric,
+                                           subject_keys = vars(STUDYID, USUBJID),
+                                           set_values_to) {
+  # Check input parameters
+  condition <- assert_filter_cond(enquo(condition))
+  assert_character_scalar(true_value)
+  assert_character_scalar(false_value)
+  assert_character_scalar(missing_value)
+  filter_add <- assert_filter_cond(enquo(filter_add), optional = TRUE)
+  assert_function(aval_fun)
+  assert_vars(subject_keys)
+  assert_data_frame(dataset_adsl, required_vars = subject_keys)
```

Comment:
assert_data_frame to be added for `dataset` too

Reply:
Added